airflow/api_fastapi/execution_api/datamodels/taskinstance.py

```diff
@@ -83,6 +83,16 @@ class TIDeferredStatePayload(BaseModel):
     next_method: str
     trigger_timeout: timedelta | None = None
 
+    @field_validator("trigger_kwargs")
+    def validate_moment(cls, v):
+        if "moment" in v and isinstance(v["moment"], str):
```

Comment:
Oh this isn't right either -- it should be called for any type I think (what if someone passes `{"moment": False}` etc - that should fail validation too.
```suggestion
        if "moment" in v:
```

Reply:
Cool, handled